src/main/java/uk/gov/companieshouse/authcodenotification/service/PrivateDataRetrievalService.java

```diff
@@ -16,26 +16,26 @@
 @Service
 public class PrivateDataRetrievalService {
 
-    private static final String OVERSEAS_ENTITY_URI_SECTION = "/overseas-entity/%s/entity-data";
+    private static final String OVERSEAS_ENTITY_URI_SECTION = "/company/%s/registered-email-address";
 
     @Autowired
     private ApiClientService apiClientService;
 
-    public OverseasEntityDataApi getOverseasEntityData(String requestId, String companyNumber)
+    public RegisteredEmailAddressJson getOverseasEntityData(String requestId, String companyNumber)
             throws ServiceException {
         var logDataMap = new DataMap.Builder().companyNumber(companyNumber).build();
         try {
             ApiLogger.infoContext(requestId, "Retrieving overseas entity data from database", logDataMap.getLogMap());
 
-            var overseasEntityDataApi = apiClientService
+            var registeredEmailAddress = apiClientService
                     .getInternalApiClient()
-                    .privateOverseasEntityDataHandler()
-                    .getOverseasEntityData(String.format(OVERSEAS_ENTITY_URI_SECTION, companyNumber))
+                    .privateCompanyResourceHandler()
+                    .getCompanyRegisteredEmailAddress(String.format(OVERSEAS_ENTITY_URI_SECTION, companyNumber))
                     .execute()
                     .getData();
 
             ApiLogger.infoContext(requestId, "Successfully retrieved overseas entity data from database", logDataMap.getLogMap());
-            return overseasEntityDataApi;
+            return registeredEmailAddress;
         } catch (ApiErrorResponseException e) {
             if (e.getStatusCode() == 404) {
                 var message = "Unable to find overseas entity data from database, HTTP exception status: " + e.getStatusCode();
```

Comment:
Here I guess the error case can also be more specific "overseas entity email address"

Reply:
The ticket requires this to be more generic so keeping "overseas entity" out of it.

Reply:
Apologies - I was looking at this through the wrong lens.